Update Box Connector with technical changes

fermujica opened the pull request.

mcherem committed:
Removed Cassandra files.

fermujica committed:
Merge branch 'latest' into DOCS-5040-mac

mcherem committed:
Dev comments for examples, & template edits.

patrik-patinak commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
-Specify access information to the connector resource.
-If a *Test* button is available, click it to test the connection.
+. Complete these fields:
+// list the fields and values
```
This list is still missing, please don't forget to update it later

patrik-patinak replied:
This list is still missing, please don't forget to update it later

kbostrom replied:
ditto: REMOVE and Jira

patrik-patinak commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
-trigger.
+
+. Complete these fields:
+// list the fields and values
```
This list is still missing, please don't forget to update it later

kbostrom replied:
GLOBALLY omit these comments from the repo and put them into Jiras.

patrik-patinak commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Create a folder called `input` under `src/main/resources directory`.
+
+. Copy the file you want to upload to the `input' directory`.
```
After the word input there is a quotation ' which shouldn't be there

patrik-patinak commented on modules/ROOT/pages/box/box-connector-get-started.adoc:
```diff
@@ -5,53 +5,55 @@ endif::[]
-<List up to 10 Exchange templates and examples for this connector -- Use the Integration
-Pattern categories (broadcast, migration, bidirectional sync, aggregation)>
+== Common Use Cases For the Connector
+// List common use cases for the connector and provide
```
Still missing list of common use cases, is this expected ?

patrik-patinak commented on modules/ROOT/pages/box/box-connector-studio.adoc:
```diff
@@ -12,15 +14,16 @@ Configure a connector in Anypoint Studio:
+. In Studio, create a Mule Project.
+. Click the Exchange *(X)* icon in the Studio task bar.
+. In Exchange, click *Login* and supply your Anypoint Platform username and password.
+. In Exchange, select *All assets* and search for \`Box\`.
```
Word Box is incorrectly formatted

mcherem committed:
Update box-connector-examples.adoc

Patrik's edits.

mcherem committed:
Update box-connector-studio.adoc

Patrik's edits.

mcherem committed:
Adding new Box template files to the nav.adoc, renamed getting started to top level box file link, addressing technical edits from Patrik P.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+= Box Connector
```
Can this topic-set title be made more specific? What *about* Box Connector? Is this an overview? does it tell the advantages of Box Connector, or what it is used to achieve? Anything else?

kbostrom replied:
Also, rework the topic-set introduction so that it...:

     Addresses the audience as "you"
     Tells exactly WHY "you" wants or needs this information, to achieve what user goal
     Mention some, if not all, of the topics represented by the next-level subtopics: prereqs., what's new, supported operations, connecting, configuring, use cases, and related info.

I know connector doc has followed one format for a long time. I will suggest to the team over Slack that the introductions be made more robust by summarizing topic set content in this way.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+include::_attributes.adoc[]
+endif::[]
+
+_Select_
```
I don't understand what this is.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+_Select_
+
+Box is an online file sharing and content management service for businesses.
```
```suggestion
Box is an online service that enables your business to share files and manage content. Anypoint Connector for Box (Box Connector) is a RAML-based connector that synchronizes this data and automates processes between Box and your CRM, CMS, mobile, and social applications.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+_Select_
+
+Box is an online file sharing and content management service for businesses.
+The Anypoint Connector for Box is a RAML based connector that provides synchronization of data and automation of processes between Box
```
Omit in favor of combining and condensing, adding "you" to the goals of using Box and the connector.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+Box is an online file sharing and content management service for businesses.
+The Anypoint Connector for Box is a RAML based connector that provides synchronization of data and automation of processes between Box
+and third party CRM, CMS, mobile, and social applications.
```
ditto

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== Prerequisites
+
+This document assumes that you are familiar with Box 2.0 API, Mule, Anypoint Connectors, Anypoint Studio, Mule concepts, elements in a Mule flow, and Global Elements.
```
```suggestion
To use this connector, you must be familiar with the Box 2.0 API, Mule runtime engine, Anypoint Connectors, Anypoint Studio, and Mule concepts, including flow elements and global elements.
```

kbostrom replied:
GLOBALLY standardize between Global Element and global element. 
I performed an Internet search, but I see both versions used.
If you can use "a" or "the" before it, then it should be lowercased.
It does not seem to me to be a real proper noun (proprietary name).
If lowercase, init. cap. only if quoting a UI label that is init. capped.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+This document assumes that you are familiar with Box 2.0 API, Mule, Anypoint Connectors, Anypoint Studio, Mule concepts, elements in a Mule flow, and Global Elements.
+
+To use this connector you need the following:
```
"You must also have:"

kbostrom replied:
No periods at the end of the two list items that follow

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+For hardware and software requirements and compatibility information, see the Connector Release Notes.
+
+
+== What's New in this Connector
```
Capitalize "This" in titles.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== What's New in this Connector
+
+This version of the connector drops support for Mule 3.x and adds support for Mule 4.x.
```
Consider "removes" instead of "drops," just for the purposes of Global English.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+This version of the connector drops support for Mule 3.x and adds support for Mule 4.x.
+
+This is a RAML based connector and only uses REST calls to interact with Box 2.0 API and only supports a subset of services available in it.
```
```suggestion
Because this is a RAML-based connector, it uses only REST calls to interact with Box 2.0 API and supports only a subset of services available in the API.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== Supported Operations
+
+The given below is a list of supported operations grouped under a category that the Box connector supports.
```
```suggestion
The following table presents supported operations grouped by categories that Box Connector supports:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|===
+|Category |Supported Operations
+
+| Collaborations a| * Create Collaboration
```
GLOBALLY change the table-cell entries for the Supported Operations column to use sentence-style capitalization:

     Create collaboration
     Delete collaboration
     Get collaboration
     Get pending collaborations....

...unless these are legitimate proper nouns.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|===
+
+
+== To Connect in Design Center
```
```suggestion
== Connect in Design Center
```

kbostrom replied:
Should we begin with some text here that tells the audience that it can connect either in DC or in Studio, and why it might choose one over the other?
This could actually serve as a transition between the title and the first step: 

"You can connect either in Anypoint Design Center (Design Center) or in Anypoint Studio. Connect in Design Center when...."

...and then have a similar intro to the Studio text.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== To Connect in Design Center
+
+. In Design Center, click a trigger such as HTTP Listener or Scheduler.
```
```suggestion
. In Design Center, click a trigger such as *HTTP Listener* or *Scheduler*.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+== To Connect in Design Center
+
+. In Design Center, click a trigger such as HTTP Listener or Scheduler.
+. Click the Plus symbol which is right after Reusable Configurations link. And to use Box connector global element, select `mule-box-connector` from the Reusable Configurations.
```
```suggestion
. Click the + symbol to the right of the Reusable Configurations link. 
. To use the Box Connector global element, select *mule-box-connector* from Reusable Configurations.
```
The original sounds like two steps, but since I cannot view the screen capture, I have guessed here.
When does the user click the link...or does clicking + open the link, and then the user selects mule-box-connector? Clarify.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+image::box/box-reusable-config.png[Configuration selector]
++
+. Click Create Configuration in the above screen.
```
```suggestion
. Click *Create Configuration*.
```
Be sure to format all UI labels of items that the user clicks or selects in boldface.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+image::box/box-reusable-config.png[Configuration selector]
++
+. Click Create Configuration in the above screen.
++
```
I suggest omitting all of the following text, up to the next step in the task.
If the user is performing this task, then that person SEES all of this onscreen.
The properties are named intuitively, so there is really no reason to define them.
You can still include the screen captures that show what happens after the previous step, if you want to, but my feeling is that this is all pretty obvious.
I would "locate" the user at the beginning of the next step; see below.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+** Authorization Url: The authorization URL to request for an authorization code.
+** Access Token Url: The access token URL to request for an access token.
++
+. Select the plus sign in the flow to add a component.
```
```suggestion
. In the resulting configuration screen for OAUTH 2.0 with Authorization Code grant type, click + in the flow to add a component.
```

kbostrom replied:
Are the previous and the following steps repeats of Step 2 earlier? Is this intentional?

kbostrom replied:
Also, please verify the UI label of the configuration screen: is it really everything from OAUTH to "grant type"?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+** Access Token Url: The access token URL to request for an access token.
++
+. Select the plus sign in the flow to add a component.
+. Select `mule-box-connector` from the list of components to get the Box connector.
```
```suggestion
. Select *`mule-box-connector`* from the list of components.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+image::box/box-select.png[Connector select]
++
+. Pick an operation. For example, Get Folder.
```
What does this mean, "pick"? Is the user making a decision or actually clicking something?

Do we mean "Select an operation: for example, **Get Folder**."

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+. Pick an operation. For example, Get Folder.
++
+. Select the `box-connector-config` configuration that had been created previously. And configure the fields:
```
```suggestion
. Select your *box-connector-config*.
. Optionally, use a comma-separated list to name the fields to include in the response.
. Provide an ID for the folder <from which? to which?> the information is <retrieved?><stored?>
```

kbostrom replied:
Use one step per user action.
Clarify what those actions are.
Clarify whether the user is creating a folder in which to store data or from which to retrieve it.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+** Fields: The specific comma-separated list of fields to include in the response. It is optional.
+** Folder Id: The ID of the folder for which the information is retrieved. It is required.
+
```
Is there anything else that needs to be said about this task completion?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+** Fields: The specific comma-separated list of fields to include in the response. It is optional.
+** Folder Id: The ID of the folder for which the information is retrieved. It is required.
+
+== To Connect in Anypoint Studio 7
```
```suggestion
== Connect in Anypoint Studio 7
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== To Connect in Anypoint Studio 7
+
+You can use this connector in Anypoint Studio by first downloading it from Exchange and configuring it as needed.
```
What does "configure as needed" mean? Should we refer to specific steps from the previous task?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+You can use this connector in Anypoint Studio by first downloading it from Exchange and configuring it as needed.
+
+=== To Install Connector in Studio
```
```suggestion
=== Install Connector in Studio
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+=== To Install Connector in Studio
+
+. In Anypoint Studio, click the Exchange icon in the Studio taskbar.
+. Click Login in Anypoint Exchange.
```
```suggestion
. Click *Login*.
```
Is "login" really capitalized in the UI?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+. In Anypoint Studio, click the Exchange icon in the Studio taskbar.
+. Click Login in Anypoint Exchange.
+. Search for the Box connector and click Install.
```
```suggestion
. Locate the Box Connector and click *Install*.
```
Can we clarify the search/locate: is the audience looking for an icon? a menu selection? a pane? a UI label?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. In Anypoint Studio, click the Exchange icon in the Studio taskbar.
+. Click Login in Anypoint Exchange.
+. Search for the Box connector and click Install.
+. Follow the prompts to install the connector.
```
Yay! I love this kind of step. Our users know how to use an interface. Uh. IF the interfaces is designed properly....

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Search for the Box connector and click Install.
+. Follow the prompts to install the connector.
+
+When Studio has an update, a message displays in the lower right corner which you can click to install the update.
```
```suggestion
Note that when Studio is updated, a message displays in the lower-right corner of the display, which you must click to install the update.
```

kbostrom replied:
How does a Studio update impact an already-installed connector? Do we need to say more here?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+When Studio has an update, a message displays in the lower right corner which you can click to install the update.
+
+=== To Configure in Studio
```
```suggestion
=== Configure Connector in Studio
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+</dependency>
+----
+
+== To Configure the Box Connector Global Element
```
```suggestion
== Configure the Box Connector Global Element
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== To Configure the Box Connector Global Element
+
+To use the Box connector in a Mule application, configure a global Box configuration element that can be used by as many Box connectors as
```
```suggestion
To use Box Connector in a Mule application, configure a global Box configuration element that can be used by as many Box connectors as
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+To use the Box connector in a Mule application, configure a global Box configuration element that can be used by as many Box connectors as
+you require for your application.
+
+. Click the Global Elements tab at the base of the canvas.
```
```suggestion
. Click the *Global Elements* tab at the bottom of the Design Center canvas.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+you require for your application.
+
+. Click the Global Elements tab at the base of the canvas.
+. In the Global Configuration Elements screen, click Create.
```
```suggestion
. In the Global Configuration Elements screen, click *Create*.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+image::box/box-config-global-wizard.png[Global Element Configuration Wizard]
++
+. In the Choose Global Type wizard, expand Connector Configuration and select Mule-box-connector Config and click Ok.
```
```suggestion
. In the Choose Global Type wizard, expand Connector Configuration, select *Mule-box-connector Config*, and click *Ok*.
```
Be sure that you are presenting UI labels exactly as they appear onscreen (capitals, hyphens, spelling, and so on).

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+image::box/box-config-global-wizard.png[Global Element Configuration Wizard]
++
+. In the Choose Global Type wizard, expand Connector Configuration and select Mule-box-connector Config and click Ok.
+The configuration screen for OAUTH 2.0 with Authorization Code grant type appears. This is the only configuration that Box connector supports.
```
```suggestion
. Configure the properties as shown:
```
Per previous comments, no need to repeat what is shown in the UI. Be sure to follow this order when using graphics (screenshots, tables, diagrams, lists, other nonnarrative elements)

1. Introduce with a sentence ending with a colon.
2. Show.
3. Explain details already shown, if necessary.

Do not present graphics first and then put all text about them after.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+image::box/box-oauth-config.png[Global Element Properties]
++
+. Configure the properties as per the image above. Given below has the description for the properties.
```
move as suggested above

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Configure the properties as per the image above. Given below has the description for the properties.
++
+[%header,cols="30a,70a"]
+|===
```
Ditto previous comment about this text, which I think is unnecessary. UI labels are self-explanatory for our audience.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|accessTokenUrl|The access token URL to request for an access token
+|===
++
+In the OAUTH 2.0 configuration screen above, the placeholder values refer to a configuration file `mule-artifact.properties` placed in the
```
```suggestion
In this screen, the placeholder values refer to a configuration file, `mule-artifact.properties`, placed in the
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|===
++
+In the OAUTH 2.0 configuration screen above, the placeholder values refer to a configuration file `mule-artifact.properties` placed in the
+`src/main/resources` folder of your project.
```
Does the user have to create this folder or is it automatically there?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+In the OAUTH 2.0 configuration screen above, the placeholder values refer to a configuration file `mule-artifact.properties` placed in the
+`src/main/resources` folder of your project.
++
+You can either enter your credentials into the global configuration properties, or reference a configuration file that contains these values. For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file. Keeping these values in a separate file is useful if you need to deploy to different environments, such as production, development, and QA, w ...
```
This task is documenting using the configuration file instead of the global config props file, so that is all we should refer to in this topic set. 
I suggest omitting the previous paragraph in the effort to document ONE "golden path" to the result we want.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+You can either enter your credentials into the global configuration properties, or reference a configuration file that contains these values. For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file. Keeping these values in a separate file is useful if you need to deploy to different environments, such as production, development, and QA, w ...
++
+. Click OK to save the global connector configurations.
```
```suggestion
. Click *OK*.
```
Note that a previous step shows OK as Ok. Please verify all UI labels, since I cannot view them.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+
+== Use Case: Create Folder, Retrieve Folder, and Delete Folder
+
+Use case to demonstrate Create Folder, Retrieve Folder, and Delete Folder operations.
```
Replace with an introduction to the topic content and why "you" should care.

kbostrom replied:
Be sure to locate the screen capture after the text of the step that it shows.
Do not introduce a task with a screen capture.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+image::box/box-use-case-flow.png[Create Retrieve Delete Folder Flow]
+
+. Create a new Mule Project in Anypoint Studio.
+. Add the below properties to the mule-artifact.properties file to hold your Box credentials and place it in the project's src/main/resources directory.
```
```suggestion
. Add the following properties to `mule-artifact.properties` and place the file in the project's `src/main/resources` directory:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+config.clientSecret=<Client Secret>
+----
++
+. Configure the HTTP Listener by adding a new HTTP Global Element. Click the plus sign next to the Connector configuration field.
```
Reword to put the user actions in chrono order, which I believe is...:

. Click + next to the connector configuration field to add a global element.    [[Does this have a UI label?]]
. Configure the global element as follows:

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+----
++
+. Configure the HTTP Listener by adding a new HTTP Global Element. Click the plus sign next to the Connector configuration field.
+. Configure the global element according to the table below:
```
Remove per above suggested replacement

kbostrom replied:
GLOBALLY replace two-column item+description tables with a definition list. See style guide.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|===
+|Parameter |Value
+|Display Name |HTTP Listener
+|Connector configuration | Select the HTTP Listener Configuration that had been created previously
```
```suggestion
|Connector configuration | The newly created HTTP Listener configuration
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Path |`/boxdemo`
+|===
++
+. Now set the folder properties for the folder we are going to create using a DataWeave component. Drag the DataWeave component next to the HTTP Listener and use the script below.
```
```suggestion
. Drag the DataWeave component next to the HTTP Listener and configure as follows:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+}
+----
++
+. Drag Create folder operation of the Box connector next to the DataWeave component to create a folder.
```
```suggestion
. Drag the connector *Create folder* operation to the DataWeave component.
```

kbostrom replied:
PLEASE VERIFY the accuracy of my suggested rewordings, because I am having trouble guessing what the original text really means. 
Again, focus on one user action per step.
Boldface the UI label that the user clicks or selects or drags.
Clarify the start of the action and the end of the action.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+----
++
+. Drag Create folder operation of the Box connector next to the DataWeave component to create a folder.
+. Configure the Box connector by adding a new Box Global Element. Click the plus sign next to the Connector configuration field.
```
```suggestion
. Add a Box global element by clicking the + next to the Connector configuration field.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+. Drag Create folder operation of the Box connector next to the DataWeave component to create a folder.
+. Configure the Box connector by adding a new Box Global Element. Click the plus sign next to the Connector configuration field.
+. Configure the global element according to the table below:
```
```suggestion
. Configure the global element as follows:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+[%header%autowidth.spread]
+|===
+|Parameter |Value |Comments
+|clientId |${config.clientId} |The client identifier as assigned by the authorization server when the client application was registered
```
```suggestion
|clientId |${config.clientId} |Specifies the client identifier as assigned by the authorization server when the client application was registered.
```
I am editing to make these complete sentences so that the items in this column will be parallel....

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|===
+|Parameter |Value |Comments
+|clientId |${config.clientId} |The client identifier as assigned by the authorization server when the client application was registered
+|clientSecret |${config.clientSecret} |The client application's client secret
```
```suggestion
|clientSecret |${config.clientSecret} |Specifies the client application's client secret.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|basePath |/2.0 |This is the default value. Leave as is.
+|protocol |HTTPS |This is the default value. Leave as is.
+|localCallbackPath |/callback |This is the default value. Leave as is.
+|localCallbackConfig |HTTP_Listener_config |Select the HTTP Listener configuration created earlier
```
```suggestion
|localCallbackConfig |HTTP_Listener_config |Select the newly created HTTP Listener configuration.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|protocol |HTTPS |This is the default value. Leave as is.
+|localCallbackPath |/callback |This is the default value. Leave as is.
+|localCallbackConfig |HTTP_Listener_config |Select the HTTP Listener configuration created earlier
+|externalCallbackUrl |`+http://localhost:8081/callback+`  |Give the redirect URL that is configured in the client application
```
```suggestion
|externalCallbackUrl |`+http://localhost:8081/callback+`  |Specifies the redirect URL that is configured in the client application.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|localCallbackPath |/callback |This is the default value. Leave as is.
+|localCallbackConfig |HTTP_Listener_config |Select the HTTP Listener configuration created earlier
+|externalCallbackUrl |`+http://localhost:8081/callback+`  |Give the redirect URL that is configured in the client application
+|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be hit after deploying the mule application to initiate OAUTH2 dance
```
```suggestion
|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |The URL to visit after deploying the Mule application to initiate the OAUTH2 dance.
```
Please verify the language used here. I tried to find an alternative to "hit." And is there one for "dance"?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|localCallbackConfig |HTTP_Listener_config |Select the HTTP Listener configuration created earlier
+|externalCallbackUrl |`+http://localhost:8081/callback+`  |Give the redirect URL that is configured in the client application
+|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be hit after deploying the mule application to initiate OAUTH2 dance
+|authorizationUrl |`+https://account.box.com/api/oauth2/authorize+`  |The authorization URL to request for an authorization code
```
```suggestion
|authorizationUrl |`+https://account.box.com/api/oauth2/authorize+`  |Specifies the authorization URL.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|externalCallbackUrl |`+http://localhost:8081/callback+`  |Give the redirect URL that is configured in the client application
+|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be hit after deploying the mule application to initiate OAUTH2 dance
+|authorizationUrl |`+https://account.box.com/api/oauth2/authorize+`  |The authorization URL to request for an authorization code
+|accessTokenUrl |`+https://api.box.com/oauth2/token+`  |The access token URL to request for an access token
```
```suggestion
|accessTokenUrl |`+https://api.box.com/oauth2/token+`  |Specifies the URL for requesting for an access token.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|accessTokenUrl |`+https://api.box.com/oauth2/token+`  |The access token URL to request for an access token
+|===
++
+The corresponding XML configuration should be as follows:
```
What is "the corresponding XML config"? Is this something that the user works with or is this the result of the step or is this a default or what? How does this relate to the step under which it appears?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+ property_externalCallbackUrl="http://localhost:8081/callback" />
+----
++
+. Back in the properties editor of the Box connector, configure the parameters required for the Create folder operation:
```
```suggestion
. In the properties editor for the connector, configure the parameters required for the Create folder operation:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Back in the properties editor of the Box connector, configure the parameters required for the Create folder operation:
++
+[%header%autowidth.spread]
+|===
```
Make this a definition list rather than a table.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Create folder request data |`#[payload]`
+|===
++
+. Drag a Logger component to log the newly created folder ID and configure the following properties.
```
Drag it where?
And what is the step: logging or configuring?

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Drag a Logger component to log the newly created folder ID and configure the following properties.
++
+[%header%autowidth.spread]
+|===
```
Make a definition list. 
GLOBAL for two-column tables.
I will stop marking this from here on.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+[%header%autowidth.spread]
+|===
+|Parameter|Value
+|Display Name|Logger to log the created Folder ID
```
Specifies the logger to use to log the created folder ID.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Parameter|Value
+|Display Name|Logger to log the created Folder ID
+2+|Generic
+|Message|Folder created with ID : `#[payload.id]`
```
```suggestion
|Message|Specifies the folder created with ID `#[payload.id]`.
```

kbostrom replied:
GLOBALLY make Value column texts parallel, as demonstrated above. I will stop marking this. Be sure to end sentences with periods.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Message|Folder created with ID : `#[payload.id]`
+|===
++
+. Drag Get folder operation of the Box connector next to the Logger component to retrieve the folder information and configure the following properties.
```
```suggestion
. Drag the Get folder operation next to the Logger component to retrieve the folder information and configure the following properties:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Folder id|`#[payload.id]`
+|===
++
+. Drag a Logger component to log the retrieved folder information and configure the following properties
```
Drag where?
End with colon.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Message|`#[payload]`
+|===
++
+. Drag Delete folder operation of the Box connector to delete the folder that is just created and configure the following properties.
```
ditto prev. edits to similar steps

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+|Folder id|`#[payload.id]`
+|===
++
+. Finally drag the DataWeave component to set the payload to display the result of the flow to the user and use the script below.
```
```suggestion
. Drag the DataWeave component to set the payload to display the result of the flow to the user as follows:
```

kbostrom replied:
Why does this Step 15 begin with "Finally" when there are 4 more steps?
This is way too many steps for one task.
If there is more than one task here, separate the topics.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+}
+----
++
+. Save and run the project as a Mule Application.
```
```suggestion
. Save and run the project as a Mule application.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+----
++
+. Save and run the project as a Mule Application.
+. Hit the URL `+http://localhost:8081/authorize+` from a browser which you had given for `localAuthorizationUrl` in global Box configuration element to initiate OAUTH2 dance.
```
GLOBALLY reword "hit the URL"
GLOBALLY reword complicated verb tenses like "which you had given" > "that you provided" to simple present, if possible, or simple past or future, if necessary. Search and fix.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Save and run the project as a Mule Application.
+. Hit the URL `+http://localhost:8081/authorize+` from a browser which you had given for `localAuthorizationUrl` in global Box configuration element to initiate OAUTH2 dance.
++
+This would display the following page asking the Box user to either grant or deny access to the client application to read and write all files and folders stored in Box.
```
This displays

I am beginning to feel as if this has not been read and cleaned up before sending to editing. Please be sure to do so going forward.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
++
+image::box/box-oauth-2-dance.png[Box OAuth 2 Dance]
++
+. Click Grant access to Box to grant the read and write permissions.
```
```suggestion
. Click *Grant access to Box*. 
```

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+. Click Grant access to Box to grant the read and write permissions.
+. Open a web browser and access the URL `+http://localhost:8081/boxdemo+`.
++
+You should get a JSON response with the following content.
```
```suggestion
. Verify task success by receiving the following message:

   `Folder created, retrieved, and deleted successfully`

You should get a JSON response with the following content.
```

kbostrom replied:
Replace line above and below per suggested replacement.

kbostrom commented on modules/ROOT/pages/box/box-connector-OLD.adoc:
```diff
@@ -0,0 +1,473 @@
+Result: `"Folder created, retrieved, and deleted successfully"`
+
+== Use Case: XML
+
```
Provide text that introduces the following syntax example.
Provide text that explains the syntax example afterward.
Or is this syntax self-explanatory?

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
 
```
The text above this, in which I cannot edit, contains internal notes.
GLOBALLY replace things like this with Jira entries; do not leave notes like this in public repo text:

// Writer: Ensure that the connector reference xref and title are correct.

kbostrom replied:
The steps below are so much better than what I saw earlier!

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
 == Configure the Target Component
 
-. Click "+" to the right of the trigger card.
+. Click `+` to the right of the trigger card.
```
Do not format this at all. Or, use bold, since it is something that the user clicks or selects. Do not use quotes or monospace.

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
+. Click `+` to the right of the trigger card.
 . In *Select a component*, search for the connector name.
 . Select the connector name.
 . Select an an operation for the connector.
```
```suggestion
. Select an operation for the connector.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
-If a *Test* button is available, click it to test the connection.
+. Complete these fields:
+// list the fields and values
+. Click the *Test* button to test the connection.
```
```suggestion
. Click *Test*.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
+. Click the *Test* button to test the connection.
 
 == Next
 Now that you have completed configuring Design Center,
```
Do not use "now" in any of our doc.

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
+// list the fields and values
+. Click the *Test* button to test the connection.
 
 == Next
```
I believe that I have suggested that this be universally removed from connector docs. Introduce tasks with "After you have <completed this task>, <perform this task>.

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -5,7 +5,9 @@ endif::[]
-== Anypoint Studio Example: Create Folder, Retrieve Folder, and Delete Folder Flow
+
+
+== Anypoint Studio Example: Folders
```
So...it looks like I spent a ton of time editing the wrong thing.
Please take a look and apply those edits to the rest of this PR.
I have to move on to other editing requests.

kbostrom commented on modules/ROOT/pages/box/box-connector-design-center.adoc:
```diff
@@ -23,25 +24,24 @@ the xref:box/box-connector-reference.adoc[Box Connector Module Documentation Ref
 . Click *Create new application*.
-. Specify a value for *Project name*.
+. Specify a value for *Project name* and click *Create*.
 . Exit from *Let's get started* by clicking *Go straight to canvas*.
```
```suggestion
. Click *Go straight to canvas*.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -154,7 +156,7 @@ The corresponding XML configuration should be as follows:
 ---
 {
-	result : "Folder got created, retrieved and deleted successfully"
+	result : "Folder was created, retrieved and deleted successfully."
```
If this is verbatim text from the UI, then we should not correct it.

```suggestion
	result : "Folder was created, retrieved, and deleted successfully."
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -167,11 +169,12 @@ This displays the following page asking the Box user to either grant or deny acc
 // image::box/box-oauth-2-dance.png[Box OAuth 2 Dance]
 +
 . Click Grant access to Box to grant the read and write permissions.
```
```suggestion
. Click *Grant access to Box*.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -167,11 +169,12 @@ This displays the following page asking the Box user to either grant or deny acc
 +
 . Click Grant access to Box to grant the read and write permissions.
+
 . Open a web browser and access the URL `+http://localhost:8081/boxdemo+`.
```
```suggestion
. Open a web browser and access http://localhost:8081/boxdemo.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -167,11 +169,12 @@ This displays the following page asking the Box user to either grant or deny acc
+
 . Open a web browser and access the URL `+http://localhost:8081/boxdemo+`.
 +
 You should get a `JSON` response with the following content.
```
```suggestion
You should get a JSON response with the following content:
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -167,11 +169,12 @@ This displays the following page asking the Box user to either grant or deny acc
 You should get a `JSON` response with the following content.
 +
-Result: `"Folder created, retrieved, and deleted successfully"`
+Result: `"Folder created, retrieved, and deleted successfully."`
```
ditto re. UI display verbatim

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -167,11 +169,12 @@ This displays the following page asking the Box user to either grant or deny acc
+Result: `"Folder created, retrieved, and deleted successfully."`
 
 
 == XML Example: Demonstrate Create Folder, Retrieve Folder, and Delete Folder operations.
```
```suggestion
== XML Example: Demonstrate Create Folder, Retrieve Folder, and Delete Folder Operations
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
 
+== Example: Upload a File
+
+This example demonstrates how to upload a file to Box.
```
This text adds no value to the title. Is there something more to say, maybe about when or why a user would upload a file to Box?

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+This example demonstrates how to upload a file to Box.
+
+. Create a new Mule Project in Anypoint Studio.
```
It might be labeled "Mule Project" in the UI, but what we are creating is "a Mule project." Does that distinction make sense?

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Create a new Mule Project in Anypoint Studio.
+
+. Create a folder called `input` under `src/main/resources directory`.
```
under > in

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Create a folder called `input` under `src/main/resources directory`.
+
+. Copy the file you want to upload to the `input directory`.
```
```suggestion
. Copy the file you want to upload to the input directory.
```
or
```suggestion
. Copy the file you want to upload to `input directory`.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Copy the file you want to upload to the `input directory`.
+
+. Configure the HTTP Listener by adding a new *HTTP Global Element*.
```
```suggestion
. Configure the HTTP Listener by adding a new HTTP global element.
```

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Configure the HTTP Listener by adding a new *HTTP Global Element*.
+
+. Click the *Global Elements* tab at the base of the canvas.
```
base > bottom

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. Click the *Global Elements* tab at the base of the canvas.
+
+. From the Global Configuration Elements screen, click *Create*.
```
From > In

kbostrom commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +253,295 @@ output application/json
+
+. From the Global Configuration Elements screen, click *Create*.
+
+. In the *Choose Global Type* wizard, expand the Connector Configuration, select HTTP Listener and click *Ok*.
```
```suggestion
. In the *Choose Global Type* wizard, expand Connector Configuration, select *HTTP Listener*, and click *Ok*.
```
Ok and not OK?

kbostrom replied:
ENDING HERE FOR NOW; will return to do more tomorrow if possible.

kbostrom reviewed (approved):
DONE for today.
I might reopen tomorrow, but unlikely. I can look again when I return to the office 16 Sept.

mcherem committed:
Update box-connector-design-center.adoc

Kim's edits.

mcherem committed:
Update box-connector-examples.adoc

Kim's edits.

mcherem committed:
Update box-connector-examples.adoc

Kim's edits.

mcherem committed:
Delete box-connector-OLD.adoc

Deleting old box connector file as it has been replaced with the new template files.

mcherem committed:
Update box-connector.adoc

mcherem committed:
Merge branch 'DOCS-5040-mac' of github.com:mulesoft/docs-connectors into DOCS-5040-mac

mcherem committed:
Update modules/ROOT/pages/box/box-connector-design-center.adoc

Kim's edits.

Co-Authored-By: Kim Bostrom <kbostrom@hortonworks.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

Kim's edits.

Co-Authored-By: Kim Bostrom <kbostrom@hortonworks.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

Kim edits.

Co-Authored-By: Kim Bostrom <kbostrom@hortonworks.com>

mcherem committed:
Update box-connector-design-center.adoc

Minor formatting fixes.

mcherem committed:
Update box-connector-examples.adoc

Formatting fixes.

mcherem committed:
Update box-connector-studio.adoc

Added backticks and spacing.

mcherem committed:
Update box-connector-config-topics.adoc

Added backticks.

mcherem committed:
Update box-connector-config-topics.adoc

Removed unused xref.

mcherem committed:
Update box-connector.adoc

Xref fixes.

mcherem committed:
Kim edits

Adding more of Kim's global edits.

mcherem committed:
Global Kim edits

Some global changes.

mcherem committed:
Global edits & procedure fix.

mcherem committed:
Update nav.adoc

Removing link to the Box Design Center template as it does not work with Box.

mcherem committed:
Update nav.adoc

Remove Box Design center link as DS does not work correctly with Box.

mcherem committed:
Merge branch 'DOCS-5040-mac' of github.com:mulesoft/docs-connectors into DOCS-5040-mac

mcherem committed:
Format display issues

mcherem committed:
Delete file and fomatting issues

Deleted Design Center template as it does not work with Box at this time.

valkyrie69 commented on modules/ROOT/pages/box/box-connector-config-topics.adoc:
```diff
@@ -4,42 +4,42 @@ include::_attributes.adoc[]
-Keeping these values in a separate file is useful if you need to deploy to different environments, such as production, development, and QA, where your access credentials differ.
 +
-. Click **OK** to save the global connector configurations.
+NOTE: For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file.
```
```suggestion
NOTE: Use a configuration file for simpler maintenance and better reusability of your project.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-config-topics.adoc:
```diff
@@ -4,42 +4,42 @@ include::_attributes.adoc[]
 +
-. Click **OK** to save the global connector configurations.
+NOTE: For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file.
+Keeping these values in a separate file is useful if you need to deploy to different environments such as: production, development, and QA (where your access credentials are different).
```
```suggestion
Keeping these values in a separate file is useful if you need to deploy to different environments where your access credentials are different, such as production, development, and QA.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -124,17 +124,17 @@ The corresponding XML configuration should be as follows:
 |===
 |Parameter|Value
-|Display Name|Logger to log the retrieved folder info.
+|Display Name|Specifies the logger to the retrieved folder info.
```
```suggestion
|Display Name|Specifies the logger that logs the retrieved folder information.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +251,300 @@ output application/json
+|Use reconnection| unchecked
+|===
++
+. Drag an HTTP request onto canvas and configure the following parameters:
```
```suggestion
. Drag an HTTP request onto the canvas and configure the following parameters:
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +251,300 @@ output application/json
++
+. Save and run the project as a Mule application.
+
+. Enter the URL `http://localhost:8081/boxUploadFile` in the browser. You should get a JSON response with the following content.
```
```suggestion
. Enter the URL `http://localhost:8081/boxUploadFile` in the browser. You should get a JSON response with the following content:
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +251,300 @@ output application/json
+
+. Enter the URL `http://localhost:8081/boxUploadFile` in the browser. You should get a JSON response with the following content.
++
+result : `"File has been successfully created"`
```
```suggestion
`result : "File has been successfully created"`
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -58,22 +58,22 @@ output application/json
+|localCallbackConfig |`HTTP_Listener_config` |Select the HTTP Listener configuration created earlier.
 |externalCallbackUrl |`+http://localhost:8081/callback+`  |Use the redirect URL that is configured in the client application.
-|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be hit after deploying the mule application to initiate `OAUTH2` dance.
+|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be entered after deploying the mule application to initiate `OAUTH2` dance.
```
```suggestion
|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |Enter this URL after deploying the Mule application to initiate `OAUTH2` dance.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-config-topics.adoc:
```diff
@@ -4,42 +4,42 @@ include::_attributes.adoc[]
+|`externalCallbackUrl`|The redirect URL that is configured in the client application.
+|`localAuthorizationUrl`|Local authorization URL that is used to initiate an `OAuth2` dance.
+|`authorizationUrl`|The authorization URL to request for an authorization code.
+|`accessTokenUrl`|Specifies the URL for requesting for an access token.
```
```suggestion
|`accessTokenUrl`|Specifies the URL for requesting an access token.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-config-topics.adoc:
```diff
@@ -4,42 +4,42 @@ include::_attributes.adoc[]
-The configuration screen for `OAUTH 2.0` with Authorization Code, grant type displays. This is the only configuration that Box connector supports.
+. In the *Global Configuration Elements* screen, click *Create*.
+. In the *Choose Global Type* wizard, expand Connector Configuration and select *Mule-box-connector Config* and click *OK*.
+. In the configuration screen for `OAUTH 2.0` with Authorization Code, grant type displays. This is the only configuration that Box connector supports.
```
This doesn't seem like a step. It seems like a step result.  I don't see the user taking any action here.  ?

valkyrie69 commented on modules/ROOT/pages/box/box-connector-config-topics.adoc:
```diff
@@ -4,42 +4,42 @@ include::_attributes.adoc[]
 :imagesdir: ../../assets/images/
 
-== Configure the Box Connector Global Element
+To use the Box connector in a Mule application, configure a global Box configuration element that can be used by as many Box connectors as needed for your application.
```
```suggestion
To use the Box connector in a Mule app, configure a global Box configuration element to use by as many Box connectors as needed for your application.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -58,22 +58,22 @@ output application/json
-|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be hit after deploying the mule application to initiate `OAUTH2` dance.
+|localAuthorizationUrl |`+http://localhost:8081/authorize+`  |This URL needs to be entered after deploying the mule application to initiate `OAUTH2` dance.
 |authorizationUrl |`+https://account.box.com/api/oauth2/authorize+`  |The authorization URL to request for an authorization code.
 |accessTokenUrl |`+https://api.box.com/oauth2/token+`  |The access token URL to request for an access token.
```
```suggestion
|accessTokenUrl |`+https://api.box.com/oauth2/token+`  |The access token URL to request an access token.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -146,35 +146,36 @@ The corresponding XML configuration should be as follows:
 ---
 {
-	result : "Folder got created, retrieved and deleted successfully"
+	result : "Folder was created, retrieved and deleted successfully."
```
This is output? I'm not sure we should be changing wording here if it's actual output. I guess we need to fix the source in this case.

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -146,35 +146,36 @@ The corresponding XML configuration should be as follows:
 ----
 +
-. Save and run the project as a Mule Application.
+. Save and run the project as a Mule application.
```
```suggestion
. Save and run the project as a Mule app.
```

valkyrie69 commented on modules/ROOT/pages/box/box-connector-examples.adoc:
```diff
@@ -250,7 +251,300 @@ output application/json
++
+. Configure the File connector by adding a new *File Global Element*.
+
+. Click the *Global Elements* tab at the base of the canvas.
```
```suggestion
. Click the *Global Elements* tab at the bottom of the canvas.
```

valkyrie69 reviewed (approved):
Minor stuff. Assume local build was done and links checked.

mcherem committed:
Update modules/ROOT/pages/box/box-connector-config-topics.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-config-topics.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-config-topics.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-config-topics.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update modules/ROOT/pages/box/box-connector-examples.adoc

V's edits.

Co-Authored-By: valkyrie69 <36547707+valkyrie69@users.noreply.github.com>

mcherem committed:
Update box-connector-config-topics.adoc

V's edit.
